File: 1-25/euler002.cpp
#include <iostream>

int main()
{
	int sum = 0;
	for (int fibo1 = 1, fibo2 = 2; fibo2 <= 4000000;) {
		if (fibo2 % 2 == 0)
			sum += fibo2;
		fibo2 += fibo1;
		fibo1 = fibo2 - fibo1;
	}
	std::cout << sum << "\n";
}

File: 1-25/euler002.rs
fn main() {
    let mut x = 0;
    let mut y = 1;
    let mut sum = 0;
    while y <= 4000000 {
        if y % 2 == 0 {
            sum += y;
        }
        let z = x;
        x = y;
        y += z;
    }
    println!("{}", sum);
}
